### PR TITLE
Support for CocoaPods 1.0

### LINKED ISSUE #137: When I want to use pod papackage to make a static library in cocoapods - 1.0.0,it has an error , but in cocoapods - 0.39.0 without error
Usage of the constant Pod::SourcesManager is deprecated, use Pod::Config.instance.sources_manager instead (called from /Users/admin/.rvm/gems/ruby-2.0.0-p598@global/gems/cocoapods-packager-1.3.0/lib/cocoapods-packager/pod_utils.rb:73:in `spec_with_name')

――― MARKDOWN TEMPLATE ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

Command

/Users/admin/.rvm/gems/ruby-2.0.0-p598/bin/pod package SDK.podspec --force
Report

What did you do?

What did you expect to happen?

What happened instead?

Stack

   CocoaPods : 1.0.0
        Ruby : ruby 2.0.0p598 (2014-11-13 revision 48408) [x86_64-darwin14.0.0]
    RubyGems : 2.6.4
        Host : Mac OS X 10.11.2 (15C50)
       Xcode : 7.3 (7D175)
         Git : git version 2.6.4 (Apple Git-63)
Ruby lib dir : /Users/admin/.rvm/rubies/ruby-2.0.0-p598/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ bd0d96bba465d6656a1abded73970d688756518d
Plugins

cocoapods-deintegrate : 1.0.0
cocoapods-packager    : 1.3.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.0.0
Error

NoMethodError - undefined method `integrate_targets=' for #<Pod::Config:0x007f9af3a20f40>
/Users/admin/.rvm/gems/ruby-2.0.0-p598@global/gems/cocoapods-packager-1.3.0/lib/pod/command/package.rb:65:in `build_in_sandbox'
/Users/admin/.rvm/gems/ruby-2.0.0-p598@global/gems/cocoapods-packager-1.3.0/lib/pod/command/package.rb:84:in `block in build_package'
/Users/admin/.rvm/gems/ruby-2.0.0-p598@global/gems/cocoapods-packager-1.3.0/lib/pod/command/package.rb:83:in `each'
/Users/admin/.rvm/gems/ruby-2.0.0-p598@global/gems/cocoapods-packager-1.3.0/lib/pod/command/package.rb:83:in `build_package'
/Users/admin/.rvm/gems/ruby-2.0.0-p598@global/gems/cocoapods-packager-1.3.0/lib/pod/command/package.rb:56:in `run'
/Users/admin/.rvm/gems/ruby-2.0.0-p598@global/gems/claide-1.0.0/lib/claide/command.rb:334:in `run'
/Users/admin/.rvm/gems/ruby-2.0.0-p598@global/gems/cocoapods-1.0.0/lib/cocoapods/command.rb:50:in `run'
/Users/admin/.rvm/gems/ruby-2.0.0-p598@global/gems/cocoapods-1.0.0/bin/pod:55:in `<top (required)>'
/Users/admin/.rvm/gems/ruby-2.0.0-p598/bin/pod:23:in `load'
/Users/admin/.rvm/gems/ruby-2.0.0-p598/bin/pod:23:in `<main>'
――― TEMPLATE END ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

[!] Oh no, an error occurred.

### PR DESCRIPTION
**Pull request by neonichu**
OK, it's finally time for this ⏳ 

Most of the changes are straightforward, a lot of it is updating the specs to work properly with CP 1.0. I also finally created a sensible rubocop configuration and made it run on CI.

Fixes #137, #130

### CODE REVIEW

**Comment by segiddins** [comment]
Should rebase to get rid of `	Merge remote-tracking branch 'origin/master'`

File: .rubocop.yml [hunk 1]
```diff
@@ -14,3 +14,21 @@ Style/Tab:
 LineLength:
   Enabled: false
 
+ClassLength:
```
Comment (segiddins) [inline .rubocop.yml]: why not just use the CocoaPods config?

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,9 @@
-osx_image: xcode7
+osx_image: xcode7.3
 language: objective-c
 cache: bundler
 before_install:
 - bundle install
 - bundle exec pod repo update --silent
-script: bundle exec rake specs
+script:
+- bundle exec rake specs
```
Comment (segiddins) [inline .travis.yml]: `rake spec` for consistency with the rest of the CocoaPods projects?

File: Gemfile.lock [hunk 3]
```diff
@@ -81,11 +74,24 @@ GEM
...
+    powerpack (0.1.1)
+    prettybacon (0.0.2)
+      bacon (~> 1.2)
+    rainbow (2.1.0)
     rake (10.4.2)
     rest-client (1.8.0)
       http-cookie (>= 1.0.2, < 2.0)
       mime-types (>= 1.16, < 3.0)
       netrc (~> 0.7)
+    rubocop (0.38.0)
```
Comment (segiddins) [inline Gemfile.lock]: `0.40` is the latest

File: .rubocop.yml (hunk 1 above)
Comment (neonichu) [inline .rubocop.yml]: I'm using it, but it is not lenient enough 😁

File: lib/cocoapods-packager/pod_utils.rb [hunk 4]
```diff
@@ -50,6 +50,26 @@ def podfile_from_spec(path, spec_name, platform_name, deployment_target, subspec
           sources.each { |s| source s }
           platform(platform_name, deployment_target)
           pod(spec_name, options)
+
+          install!('cocoapods', :integrate_targets => false)
```
Comment (segiddins) [inline lib/cocoapods-packager/pod_utils.rb]: might also make sense to disable deterministic UUIDs?

File: lib/cocoapods-packager/pod_utils.rb [hunk 5]
```diff
@@ -50,6 +50,26 @@ def podfile_from_spec(path, spec_name, platform_name, deployment_target, subspec
...
+              if subspecs
+                subspecs.each do |subspec|
+                  pod spec_name + '/' + subspec, :podspec => path
+                end
+              else
+                pod spec_name, :podspec => path
+              end
+            elsif subspecs
+              subspecs.each do |subspec|
+                pod spec_name + '/' + subspec, :path => '.'
```
Comment (segiddins) [inline lib/cocoapods-packager/pod_utils.rb]: why not use the `:subspecs` DSL option to get rid of the nested conditionals?

**Comment by neonichu** [comment]
Interesting, the initial build broke because of stale Travis caches  ¯\\\_(ツ)\_/¯

File: lib/cocoapods-packager/pod_utils.rb (hunk 5 above)
Comment (neonichu) [inline lib/cocoapods-packager/pod_utils.rb]: Good point, I think I didn't know about them when initially writing this.

**Comment by neonichu** [comment]
:shipit:

**Pull request merged**